### ISSUE TITLE
Can't compile shader_version

### ISSUE THREAD

**Issue by ghost** [yet-another-cafebabe/osu-rs#1]
Im getting this error when trying to compile the program:
   Compiling shader_version v0.1.0
/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/shader_version-0.1.0/src/lib.rs:37:5: 37:90 error: the trait bound `Self: std::marker::Sized` is not satisfied [E0277]
/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/shader_version-0.1.0/src/lib.rs:37     fn pick_shader<'a, S: ?Sized>(self, shaders: &Shaders<'a, Self, S>) -> Option<&'a S>;
                                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/shader_version-0.1.0/src/lib.rs:37:5: 37:90 help: run `rustc --explain E0277` to see a detailed explanation
/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/shader_version-0.1.0/src/lib.rs:37:5: 37:90 help: consider adding a `where Self: std::marker::Sized` bound
/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/shader_version-0.1.0/src/lib.rs:37:5: 37:90 note: required by `Shaders`
error: aborting due to previous error
Build failed, waiting for other jobs to finish...
error: Could not compile `shader_version`.
Any idea?

**Comment by Cherrionella** [comment]
Right now i have no idea, cuz i dropped this repo about a year ago. Sorry. I'd like to continue development soon.

**Comment by ghost** [comment]
Yeah I see I just wanted to compile it and try it - I would even help but I don't have any knowledge of Rust.

**Comment by C0D3-M4513R** [comment]
I got the compiling working again. The dependencies were a real mess! (Still are) (And they are now ancient)
The game was for me only a black screen.
See #2 

> Right now i have no idea, cuz i dropped this repo about a year ago. Sorry. I'd like to continue development soon.

Soooooooon. Alright. That's always the problem. Maybe I'll make something with vulkan from scratch, with little dependencies.
